redact_test.go

```diff
@@ -8,63 +8,167 @@ import (
 )
 
 func TestSanitizeFixture(t *testing.T) {
-	gunit.Run(new(SanitizeFixture), t)
+	gunit.Run(new(SanitizeFixture), t, gunit.Options.AllSequential())
 }
 
 type SanitizeFixture struct {
 	*gunit.Fixture
 }
 
-func (this *SanitizeFixture) TestRedactDOB() {
-	input := "Hello my name is John, my date of birth is 11/1/2000 and my employee's date of birth is 01-01-2001, oh also November 1, 2000, May 23, 2019, 23 June 1989, Sept 4, 2010."
-	expectedOutput := "Hello my name is John, my date of birth is [DOB REDACTED] and my employee's date of birth is [DOB REDACTED], oh also [DOB REDACTED], [DOB REDACTED], [DOB REDACTED], [DOB REDACTED]."
+func (this *SanitizeFixture) Teardown() {
+	used = make(map[int]struct{})
```

Comment:
I'm concerned about the need to reset this global (package-level) mutable state in order for the tests to pass.

Reply:
THIS is a very good catch, adding an additional TODO... We can reset the map after the matches slice has finished populating as it will no longer be necessary at that point.

Reply:
Properly managing this mutable, package-level state to allow for concurrent usage of this library from multiple calling goroutines will require something like a `sync.Mutex`. I'd recommend moving away from mutable, package-level state of any kind. Any functions that require this book-keeping structure should initialize a fresh, local variable and pass it wherever needed.